packages/adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -0,0 +1,469 @@
+  final List<NavigationDestination>? destinations;
+
+  /// The index to be used by the [NavigationRail] if applicable.
+  final int? selectedIndex;
+
+  /// Widget to be displayed in the body slot at the smallest breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [body] is
+  /// displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? smallBody;
+
+  /// Widget to be displayed in the body slot at the middle breakpoint.
+  ///
+  /// The default displayed body.
+  final WidgetBuilder? body;
+
+  /// Widget to be displayed in the body slot at the largest breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [body] is
+  /// displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? largeBody;
+
+  /// Widget to be displayed in the secondaryBody slot at the smallest
+  /// breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [secondaryBody]
+  /// is displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? smallSecondaryBody;
+
+  /// Widget to be displayed in the secondaryBody slot at the middle breakpoint.
+  ///
+  /// The default displayed secondaryBody.
+  final WidgetBuilder? secondaryBody;
+
+  /// Widget to be displayed in the seconaryBody slot at the smallest
+  /// breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [secondaryBody]
+  /// is displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? largeSecondaryBody;
+
+  /// Defines the fractional ratio of body to the secondaryBody.
+  ///
+  /// For example 0.3 would mean body takes up 30% of the available space and
+  /// secondaryBody takes up the rest.
+  ///
+  /// If this value is null, the ratio is defined so that the split axis is in
+  /// the center of the screen.
+  final double? bodyRatio;
+
+  /// The breakpoint defined for the small size, associated with mobile-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.small].
+  final Breakpoint smallBreakpoint;
+
+  /// The breakpoint defined for the medium size, associated with tablet-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.mediumBreakpoint].
+  final Breakpoint mediumBreakpoint;
+
+  /// The breakpoint defined for the large size, associated with desktop-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.largeBreakpoint].
+  final Breakpoint largeBreakpoint;
+
+  /// Whether or not the developer wants the smooth entering slide transition on
+  /// secondaryBody.
+  ///
+  /// Defaults to true.
+  final bool internalAnimations;
+
+  /// Whether to orient the body and secondaryBody in horizontal order (true) or
+  /// in vertical order (false).
+  ///
+  /// Defaults to true.
+  final bool horizontalBody;
```

Comment:
This should be an `Axis` instead of a bool, and probably renamed to something like `bodyAxis` or `bodyOrientation`.